modules/olm-accessing-images-private-registries.adoc

```diff
@@ -0,0 +1,173 @@
+[source,json]
+----
+{
+        "auths": {
+                "registry.redhat.io": {
+                        "auth": "FrNHNydQXdzclNqdg=="
+                }
+        }
+}
+----
++
+.File storing credentials for another registry
+[source,json]
+----
+{
+        "auths": {
+                "quay.io": {
+                        "auth": "Xd2lhdsbnRib21iMQ=="
+                }
+        }
+}
+----
+--
+
+.. Create a secret that contains the authentication credentials for a private registry:
++
+[source,terminal]
+----
+$ oc create secret generic <secret_name> \
+    --from-file=.dockerconfigjson=<path/to/registry/credentials> \
+    --type=kubernetes.io/dockerconfigjson
+----
++
+Repeat this step to create additional secrets for any other required private registries, updating the `--from-file` flag to specify another registry credentials file path.
+
+. Create or update an existing `CatalogSource` object to reference one or more secrets:
++
+[source,yaml]
+----
+apiVersion: operators.coreos.com/v1alpha1
+kind: CatalogSource
+metadata:
+  name: my-operator-catalog
+  namespace: openshift-marketplace
+spec:
+  sourceType: grpc
+  secrets: <1>
+  - "<secret_name_1>"
+  - "<secret_name_2>"
+  image: <registry>:<port>/<namespace>/<image>:<tag>
+  displayName: My Operator Catalog
+  publisher: <publisher_name>
+  updateStrategy:
+    registryPoll:
+      interval: 30m
+----
+<1> Add a `spec.secrets` section and specify any required secrets.
+
+. If any Operator or Operand images that are referenced by a subscribed Operator require access to a private registry, you can either provide access to all namespaces in the cluster, or target individual namespaces.
+
+* To provide access to all namespaces in the cluster, add authentication details to the global cluster pull secret in the `openshift-config` namespace.
++
+[WARNING]
+====
+Cluster resources must adjust to the new global pull secret, which can temporarily limit the usability of the cluster.
+====
+
+.. Extract the `.dockerconfigjson` file from the global pull secret:
++
+[source,terminal]
+----
+$ oc extract secret/pull-secret -n openshift-config --confirm
+----
+
+.. Update the `.dockerconfigjson` file with your authentication credentials for the required private registry or registries and save it as a new file:
++
+[source,terminal]
+----
+$ cat .dockerconfigjson | jq --compact-output \
+    '.auths["<registry>:<port>/<namespace>/"] \//<1>
+    |= . + {"auth":"<token>"}' \//<2>
+    > new_dockerconfigjson
+----
+<1> Replace `<registry>:<port>/<namespace>` with the private registry details.
+<2> Replace `<token>` with your authentication credentials.
+
+.. Update the global pull secret with the new file:
++
+[source,terminal]
+----
+$ oc set data secret/pull-secret -n openshift-config \
+    --from-file=.dockerconfigjson=new_dockerconfigjson
+----
+
+* To update individual namespaces, add your pull secret to the `default` service account in each namespace that you want to have access:
```

Comment:
There are two problems here: 

1. The SA cannot link the `secret` from another namespace, so it requires the user to recreate the `secret` in this individual namespace. details: https://bugzilla.redhat.com/show_bug.cgi?id=1925377
2. this secret should be linked to the operator's SA, not the `default`. For example, etcd operator, it should be: `oc secret link etcd-operator secret-cs --for=pull`

Reply:
@jianzhangbjz Updated the step:

http://file.rdu.redhat.com/~adellape/012021/olm_pullsecret/operators/admin/olm-managing-custom-catalogs.html#olm-accessing-images-private-registries_olm-managing-custom-catalogs

Reply:
@adellape I didn't find any updates on this pre-review doc.